### PR TITLE
Portfolio value at date

### PR DESCRIPTION
**Pull request by adri0**
- Use downloaded quotes to calculate total portfolio value at a given date.  
- Update download_quotes to optionally download exchange rates as well.
- Update adjust_symbol function to also adjust the name of symbol if necessary.
- Update overview report to include portfolio value view.
- Update df_quotes to save same symbols as in df_tx, instead of adjusted symbols.

### CODE REVIEW

File: investd/reports/overview.py [hunk 1]
```diff
@@ -27,13 +28,20 @@
 df_tx = load_transactions()
 df_tx = df_tx[df_tx["timestamp"] <= now]
 
-# %% [markdown]
-# ### Invested amount
+# %%
+
+df_quotes = load_quotes()
+df_portfolio = views.portfolio_value(
+    df_tx, df_quotes, at_date=date(2022, 12, 30)
```
Comment (mkopec87) [inline investd/reports/overview.py:35]: hmm why not "now" as at_date?

File: investd/reports/overview.py [hunk 2]
```diff
@@ -27,13 +28,20 @@
...
+).round(2)
+
+display(df_portfolio)
 
 # %%
 Markdown(
     f"""
-{views.total_invested_ref_currency(df_tx):.2f} {INVESTD_REF_CURRENCY}
+Portfolio value at date {INVESTD_REF_CURRENCY}: {df_portfolio[f"Amount at date {INVESTD_REF_CURRENCY}"].sum()}
```
Comment (mkopec87) [inline investd/reports/overview.py:43]: maybe print the date as well?

File: investd/views.py [hunk 3]
```diff
@@ -109,3 +110,50 @@ def amount_over_time(df_tx: pd.DataFrame, period: str) -> pd.DataFrame:
...
+    )
+    df_portfolio["amount_at_date_ref_currency"] = (
+        df_portfolio["amount_at_date"] * exchange_rate_to_ref_cur
+    )
+    df_portfolio = df_portfolio.rename(
+        {
+            "amount_at_date": "Amount at date",
+            "amount_at_date_ref_currency": f"Amount at date {INVESTD_REF_CURRENCY}",
+            "amount_signed": "Invested amount",
+            "amount_ref_currency_signed": f"Invested amount {INVESTD_REF_CURRENCY}",
```
Comment (mkopec87) [inline investd/views.py:154]: maybe we can be consistent in appending "_ref_currency "either at the end or in the middle?

**Review by mkopec87 — CHANGES_REQUESTED** [review]
I think the report date should be dynamic, the other comments are optional :)

File: investd/reports/overview.py [hunk 4]
```diff
@@ -76,3 +77,14 @@
...
+# %%
+
+df_quotes = load_quotes()
+df_portfolio = views.portfolio_value(
+    df_tx, df_quotes, at_date=date(2022, 12, 30)
+).round(2)
+
+display(df_portfolio)
+
+print("Total amount PLN:", df_portfolio[f"Amount at date {INVESTD_REF_CURRENCY}"].sum())
```
[the remark this answers is not in the document]
Comment (adri0) [reply investd/reports/overview.py]: Added the total row :)

File: investd/reports/overview.py (hunk 1 above)
Comment (adri0) [reply investd/reports/overview.py]: Added instead a `report_date` parameter that can be provided in the command line.

File: investd/reports/overview.py (hunk 2 above)
Comment (adri0) [reply investd/reports/overview.py]: Check if with the report date it's clear now

File: investd/views.py (hunk 3 above)
Comment (adri0) [reply investd/views.py]: Fixed. I prefer leaving the signed always at the end, so moved the _ref_currency to stay next to "amount"

**Pull request merged**